pkg/spec/spec.go

```diff
@@ -360,6 +356,11 @@ func (config *CreateConfig) createConfigToOCISpec(runtime *libpod.Runtime, userM
 	if err := config.Cgroup.ConfigureGenerator(&g); err != nil {
 		return nil, err
 	}
+
+	config.Env = env.Join(defaultEnv, config.Env)
+	for name, val := range config.Env {
+		g.AddProcessEnv(name, val)
+	}
```

Comment:
It might be good to add an inline comment here to explain why it's here to make sure it doesn't get moved up again inadvertently.

Reply:
This is a back port.  And in code that is not used in the new version. I added a test for this situation in the master branch pull.  I think we need more tests to make sure that environment handling is correct.